bin/copyRawDataToTmp.sh

```diff
@@ -0,0 +1,356 @@
+
+#
+# Use multiplexing to reduce the amount of SSH connections created
+# when rsyncing using the group's data manager account.
+# 
+#  1. Become the "${DATA_MANAGER} user who will rsync the data to prm and 
+#  2. Add to ~/.ssh/config:
+#		ControlMaster auto
+#		ControlPath ~/.ssh/tmp/%h_%p_%r
+#		ControlPersist 5m
+#  3. Create ~/.ssh/tmp dir:
+#		mkdir -p -m 700 ~/.ssh/tmp
+#  4. Recursively restrict access to the ~/.ssh dir to allow only the owner/user:
+#		chmod -R go-rwx ~/.ssh
+#
+
+#
+# Get a list of all samplesheets for this group on the specified samplesheetsServerLocation, where the raw data was generated, and
+#	1. Loop over their analysis ("run") sub dirs and check if there are any we need to rsync.
+#	2. Optionally, split the samplesheets per project after the data was rsynced.
+#
+# shellcheck disable=SC2029
+readarray -t sampleSheetsFolder < <(ssh "${samplesheetsServerLocation}" "find \"${samplesheetsLocation}/\" -mindepth 1 -maxdepth 1 -type f -name '*.${SAMPLESHEET_EXT}'")
+
+##ISSUE: this script should always be executed by the ateambot user (and not for chaperone with the dm user solely to write the logs)
+if [[ "${#sampleSheetsFolder[@]:-0}" -eq '0' ]]
+then
+	log4Bash 'WARN' "${LINENO}" "${FUNCNAME:-main}" '0' "No samplesheets found at ${samplesheetsServerLocation} ${samplesheetsLocation}/*.${SAMPLESHEET_EXT}."
+else
+	for sampleSheet in "${sampleSheetsFolder[@]}"
+	do
+		#
+		# Process this samplesheet / run 
+		#
+		project="$(basename "${sampleSheet%."${SAMPLESHEET_EXT}"}")"
+		controlFileBase="${WORKING_DIR}/logs/${project}/"
+		export JOB_CONTROLE_FILE_BASE="${controlFileBase}/${project}.${SCRIPT_NAME}"
+		printf '%s\n' "Rsyncing ${project} dir ..." >> "${lockFile}"
+		# shellcheck disable=SC2174
+		mkdir -m 2770 -p "${WORKING_DIR}/logs/${project}/"
+		if [[ -e "${JOB_CONTROLE_FILE_BASE}.finished" ]]
+		then
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Skipping already processed run ${project}."
+			continue
+		else
+			log4Bash 'INFO' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Processing run ${project} ..."
+		fi
+		
+		printf '' > "${JOB_CONTROLE_FILE_BASE}.started"
+		
+		if [[ "${samplesheetsServerLocation}" == "localhost" ]]
+		then
+			log4Bash 'TRACE' "${LINENO}" "${FUNCNAME[0]:-main}" '0' "Samplesheet is on this machine"
+			if [[ -f "${WORKING_DIR}/logs/${project}/${RAWDATAPROCESSINGFINISHED}" ]]
+			then
+				log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "${WORKING_DIR}/logs/${project}/${RAWDATAPROCESSINGFINISHED} present."
+				##Check if array or NGS run
+				# shellcheck disable=SC2029
+				if ssh "${samplesheetsServerLocation}" "grep 'SentrixBarcode_A' \"${sampleSheet}\""
+				then
+					log4Bash 'TRACE' "${LINENO}" "${FUNCNAME:-main}" '0' "this is array data"
+					## copy all the data from that run
+					rsync -rlptDvc "${WORKING_DIR}/./rawdata/array/GTC/${filePrefix}" "${DESTINATION_DIAGNOSTICS_CLUSTER}:${TMP_ROOT_DIR}/"
```

Comment:
The `/./` in the source path will not work without `--relative`

Reply:
This `rsync` on `localhost` does not need the `--rsync-path` with `sudo` construct if this script is executed as `*-ateambot` user, but we currently ensure it is always executed with the data manager account using this block of code:
```
#
# Write access to prm storage requires data manager account.
#
if [[ "${ROLE_USER}" != "${DATA_MANAGER}" ]]; then
	log4Bash 'FATAL' "${LINENO}" "${FUNCNAME:-main}" '1' "This script must be executed by user ${DATA_MANAGER}, but you are ${ROLE_USER} (${REAL_USER})."
fi
```
So I don't think this will work on localhost. We should make that check smarter and define something like a data staging user variable, which can be `${DATA_MANAGER}` on a _chaperone_ machine and `${ATEAMBOTUSER}` on localhost.

Reply:
fixed in lines 202-213